cpp/daal/src/data_management/finiteness_checker.cpp

```diff
@@ -158,6 +230,68 @@ double computeSum<double, avx512>(size_t nDataPtrs, size_t nElementsPerPtr, cons
     return computeSumAVX512Impl<double>(nDataPtrs, nElementsPerPtr, dataPtrs);
 }
 
+double computeSumSOAAVX512Impl(NumericTable & table, bool & sumIsFinite)
+{
+    SafeStatus safeStat;
+    double sum                                  = 0;
+    bool breakFlag                              = false;
+    const size_t nRows                          = table.getNumberOfRows();
+    const size_t nCols                          = table.getNumberOfColumns();
+    NumericTableDictionaryPtr tableFeaturesDict = table.getDictionarySharedPtr();
+
+    daal::TlsMem<double, avx512, services::internal::ScalableCalloc<double, avx512> > tlsSum(1);
+    daal::TlsMem<bool, avx512, services::internal::ScalableCalloc<bool, avx512> > tlsFinite(1);
+    daal::threader_for(nCols, nCols, [&](size_t i) {
+        double * localSum  = tlsSum.local();
+        bool * localFinite = tlsFinite.local();
+
+        switch ((*tableFeaturesDict)[i].getIndexType())
+        {
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT32:
+        {
+            ReadColumns<float, avx512> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS_THR(colBlock);
+            const float * colPtr = colBlock.get();
+            *localSum += computeSum<float, avx512>(1, nRows, &colPtr);
+            break;
+        }
+        case daal::data_management::features::IndexNumType::DAAL_FLOAT64:
+        {
+            ReadColumns<double, avx512> colBlock(table, i, 0, nRows);
+            DAAL_CHECK_BLOCK_STATUS_THR(colBlock);
+            const double * colPtr = colBlock.get();
+            *localSum += static_cast<double>(computeSum<double, avx512>(1, nRows, &colPtr));
+            break;
+        }
+        }
+
+        *localFinite |= valuesAreNotFinite(localSum, 1, false);
+        if (*localFinite)
+        {
+            breakFlag = true;
+            daal::break_threader_for();
+        }
+    });
+    tlsSum.reduce([&](double * localSum) { sum += *localSum; });
+
+    if (breakFlag)
+    {
+        sumIsFinite = false;
+    }
+    else
+    {
+        tlsFinite.reduce([&](bool * localFinite) { sumIsFinite &= !*localFinite; });
```

Comment:
Make sure that threading works correctly by running tests and adding NaN value in any column of numpy ndarray.